parmed/tools/add1264.py

```diff
@@ -78,89 +78,88 @@ def params1264(parm, mask, c4file, watermodel, polfile, tunfactor):
 
     print("***********************************************************")
     # Determine which atom type was treated as the center metal ion
-    typelist = dict()
+    mettypdict = dict()
     for i in mask.Selected():
-        typeidx = parm.parm_data['ATOM_TYPE_INDEX'][i]
-        chgi = parm.parm_data['CHARGE'][i]
-        if typeidx in typelist: continue
-        typelist[typeidx] = (parm.atoms[i].atomic_number, int(chgi))
-        print("The selected metal ion is",
-              pt.Element[parm.atoms[i].atomic_number])
-    types = typelist.keys()
-    types.sort()
-
-    # Determine the C4 term between the atom type of center metal ion 
-    # and every atom type in the prmtop file 
-
-    # 1. Get the list of AMBER_ATOM_TYPE and ATOM_TYPE_INDEX for all 
+        mettypind = parm.parm_data['ATOM_TYPE_INDEX'][i]
+        metchg = parm.parm_data['CHARGE'][i]
+        if mettypind in mettypdict: continue
+        mettypdict[mettypind] = (parm.atoms[i].atomic_number, int(metchg))
+        print "The selected metal ion is", \
+                pt.Element[parm.atoms[i].atomic_number]
+    mettypinds = mettypdict.keys()
+    mettypinds.sort()
```

Comment:
This doesn't work for Python 3, either.  Unlike Python 2, Python 3 does not return a list with `dict.keys()` -- it returns a more efficient `dict_keys` iterator that consumes less memory.

Change these two lines into a single line:

```Python
    mettypinds = sorted(mettypdict.keys())
```